src/main/java/entity/Monster/MonsterPower.java

```diff
@@ -0,0 +1,17 @@
+package entity.Monster;
+
+public abstract class MonsterPower implements Power{
+    /** Whether the Monster must be beaten to use this Power. True iff the Monster must be beat, false otherwise. */
+    private final boolean mustBeat;
```

Comment:
Are we doing something where the Player receives a monster's Power when they defeat them? If mustBeat is false, how else does the Player receives the power? Also how is this information going to be told to the Player

Reply:
ok nevermind i see that it's for the Steal class, but is this class really necessary? What other monster powers will need this class

Reply:
I am making the different Monster powers into different classes since usePower is implemented differently for each type of power

Reply:
Although this would not cover the scenario where a power is used regardless of whether the Monster lives or dies, I currently can't think of a logical power that would do this (certainly a Monster cant steal when it is dead).

Reply:
Ok that makes sense! Although, I think this can be confusing for the Player. They could think like "Why did I lose health? I defeated the monster?" So just keep in mind to that we should put explanations in the after fight messages. Also, it may be a good idea to have a Monster Book/List, where once a Player defeats a monster, information about the defeated monster appears and the Player read it over.